templates/pre-job.j2

```diff
@@ -1,11 +1,44 @@
 #!/usr/bin/env bash
 
-GITHUB_SOURCE_REPOSITORY=$(cat "${GITHUB_EVENT_PATH}" | jq -r '.pull_request.head.repo.full_name')
-
-# Request repo-policy-compliance service check.
-curl --noproxy '*' \
-    --fail-with-body \
-    -H 'Authorization: Bearer {{one_time_token}}' \
-    -H 'Content-Type: application/json' \
-    -d "{\"repository_name\": \"${GITHUB_REPOSITORY}\", \"source_repository_name\": \"${GITHUB_SOURCE_REPOSITORY}\", \"target_branch_name\": \"${GITHUB_BASE_REF}\", \"source_branch_name\": \"${GITHUB_HEAD_REF}\", \"commit_sha\": \"${GITHUB_SHA}\"}" \
-    http://{{host_ip}}:8080/check-run
+# Workflow dispatch - Request repo-policy-compliance service check:
+if [[ "${GITHUB_EVENT_NAME}" == "workflow_dispatch" ]]; then
+
+  logger -s "GITHUB_EVENT_NAME: ${GITHUB_EVENT_NAME}, \
+  GITHUB_REPOSITORY: ${GITHUB_REPOSITORY}, \
+  GITHUB_REF_NAME: ${GITHUB_REF_NAME}, \
+  GITHUB_SHA: ${GITHUB_SHA}"
+
+  curl --noproxy '*' \
+      --fail-with-body \
+      -H 'Authorization: Bearer {{one_time_token}}' \
+      -H 'Content-Type: application/json' \
+      -d "{\"repository_name\": \"${GITHUB_REPOSITORY}\", \"branch_name\": \"${GITHUB_REF_NAME}\", \"commit_sha\": \"${GITHUB_SHA}\"}" \
+      http://{{host_ip}}:8080/workflow_dispatch/check-run
+
+# Pull request - Request repo-policy-compliance service check:
+elif [[ "${GITHUB_EVENT_NAME}" ==  "pull_request" ]]; then
+
+  GITHUB_SOURCE_REPOSITORY=$(cat "${GITHUB_EVENT_PATH}" | jq -r '.pull_request.head.repo.full_name')
+
+  logger -s "GITHUB_EVENT_NAME: ${GITHUB_EVENT_NAME}, \
+  GITHUB_SOURCE_REPOSITORY: ${GITHUB_SOURCE_REPOSITORY}, \
+  GITHUB_REPOSITORY: ${GITHUB_REPOSITORY}, \
+  GITHUB_SOURCE_REPOSITORY: ${GITHUB_SOURCE_REPOSITORY} \
+  GITHUB_BASE_REF: ${GITHUB_BASE_REF}, \
+  GITHUB_HEAD_REF: ${GITHUB_HEAD_REF}, \
+  GITHUB_SHA: ${GITHUB_SHA}"
+
+  curl --noproxy '*' \
+      --fail-with-body \
+      -H 'Authorization: Bearer {{one_time_token}}' \
+      -H 'Content-Type: application/json' \
+      -d "{\"repository_name\": \"${GITHUB_REPOSITORY}\", \"source_repository_name\": \"${GITHUB_SOURCE_REPOSITORY}\", \"target_branch_name\": \"${GITHUB_BASE_REF}\", \"source_branch_name\": \"${GITHUB_HEAD_REF}\", \"commit_sha\": \"${GITHUB_SHA}\"}" \
+      http://{{host_ip}}:8080/check-run
+
+else
+
+  logger -p user.error -s "${GITHUB_EVENT_NAME} is not handled."
```

Comment:
Perhaps `is not supported yet, please request it to be added on <link to create issue on the GitHub runner repo>`?